src/legacy/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -880,9 +880,25 @@ function discoverController(
   };
 
   $scope.updateDataSource = Promise.method(function updateDataSource() {
-    $scope.searchSource
+    const { indexPattern, searchSource } = $scope;
+    const { timeFieldName } = indexPattern;
+    const sort = getSort($state.sort, indexPattern)
+      .map(sortPair => {
+        if (indexPattern.isTimeNanosBased() && sortPair[timeFieldName]) {
+          // for sorting on indices where the same field can be date or date_nanos
+          // it's necessary to add the numeric_type
+          return ({
+            [timeFieldName]: {
+              order: sortPair[timeFieldName],
+              numeric_type: 'date_nanos'
+            }
+          });
```

Comment:
Could we move this code into the `getSort` function itself? `getSort` is used in the search panel in Dashboard as well so I suspect putting this in the Discover controller means the sorting still won't work in Dashboard.

Reply:
Thanks Bargs, yes this needs to be adapted , I already did . I initially built this in the `getSort` function, but this broke functionality like changing the sort order in the doc table. So I thought, better add this change before setting the `sort`  of `searchSource`, since it's only relevant for Elasticsearch. Now I've extracted this to a separate function `getSortForSearchSource`, adapted also for dashboard. works

Reply:
dear @Bargs, is that change ok for you? merci!

Reply:
Sorry, I could have sworn I left a comment on this right after you made the changes. Must have forgot to hit submit. Yeah, LGTM

Reply:
great, thx!